subprojects/core/src/main/java/org/gradle/execution/taskgraph/DefaultTaskExecutionPlan.java

```diff
@@ -778,17 +818,17 @@ private String findFirstOverlap(Iterable<String> paths1, Iterable<String> paths2
         return null;
     }
 
-    private Set<String> getOutputPaths(TaskInfo task) {
+    private Set<String> getOutputPaths(TaskInfo task, FileCollection outputFiles, FileCollection localStateFiles) {
         try {
-            return canonicalizedPaths(canonicalizedFileCache, Iterables.concat(task.getOutputs(), task.getLocalState()));
+            return canonicalizedPaths(canonicalizedFileCache, Iterables.concat(outputFiles, localStateFiles));
         } catch (ResourceDeadlockException e) {
             throw new IllegalStateException("A deadlock was detected while resolving the task outputs for " + task.getTask().getIdentityPath() + ".  This can be caused, for instance, by a task output causing dependency resolution.", e);
         }
 
     }
 
-    private Set<String> getDestroyablePaths(TaskInfo task) {
-        return canonicalizedPaths(canonicalizedFileCache, task.getDestroyables());
+    private Set<String> getDestroyablePaths(FileCollection destroyableFiles) {
+        return canonicalizedPaths(canonicalizedFileCache, destroyableFiles);
     }
```

Comment:
How many of these methods can we make `static`?

Reply:
Made some methods static.